File: LeetCode/Python/13.Roman_to_integer.py
class Solution:
    def romanToInt(self, s: str) -> int:
        if s is None:
            return 0
        R = {
            "I": 1,
            "V": 5,
            "X": 10,
            "L": 50,
            "C": 100,
            "D": 500,
            "M": 1000
            }
        if s[0] in R and len(s) == 1:
            return R[s[0]]

        sum = 0
        prev = R[s[0]]
        for char in s[1:]:
            curr = R[char]
            if prev >= curr:
                sum += prev
            else:
                sum -= prev
            prev = curr
        sum += curr
        return sum

File: LeetCode/C_lang/13.RomanToInteger.c
#include "problems.h"
int romanToInt(char* s) {
    int sum = 0;
    int i = 0;
    while (s[i] != '\0') {
        if (s[i] == 'I' && s[i+1] == 'V') {
            sum += 4;
            i += 2;
        } else if (s[i] == 'I' && s[i+1] == 'X') {
            sum += 9;
            i += 2;
        } else if (s[i] == 'X' && s[i+1] == 'L') {
            sum += 40;
            i += 2;
        } else if (s[i] == 'X' && s[i+1] == 'C') {
            sum += 90;
            i += 2;
        } else if (s[i] == 'C' && s[i+1] == 'D') {
            sum += 400;
            i += 2;
        } else if (s[i] == 'C' && s[i+1] == 'M') {
            sum += 900;
            i += 2;
        } else {
            switch (s[i]) {
                case 'I':
                    sum += 1;
                    break;
                case 'V':
                    sum += 5;
                    break;
                case 'X':
                    sum += 10;
                    break;
                case 'L':
                    sum += 50;
                    break;
                case 'C':
                    sum += 100;
                    break;
                case 'D':
                    sum += 500;
                    break;
                case 'M':
                    sum += 1000;
                    break;
                default:
                    // Handle invalid input characters
                    printf("Invalid input character: %c\n", s[i]);
                    return 0;
            }
            i++;
        }
    }
    return sum;
}